charts/agent-stack-k8s/templates/deployment.yaml.tpl

```diff
@@ -11,6 +11,7 @@ spec:
     metadata:
       labels:
         app: {{ .Release.Name }}
+        {{- toYaml $.Values.labels | nindent 8 }}
```

Comment:
If a user sets
```
labels:
  app: foo
```
then the selector will no longer match the label. I think this is a footgun we should not allow.

I recommend you merge `app: {{ .Release.Name }}` back into `$.Values.labels` to prevent this, and document that the `app` label will be overwritten if it is specified by the user.

To implement this, you most likely will need to create a `_helpers.tpl` file and make some helpers in it.

Have a look at what the helm chart that is generated when you run
```
helm create agent-stack-k8s
```
does.

Reply:
> If a user sets
> 
> ```
> labels:
>   app: foo
> ```
> 
> then the selector will no longer match the label. I think this is a footgun we should not allow.
> 
> I recommend you merge `app: {{ .Release.Name }}` back into `$.Values.labels` to prevent this, and document that the `app` label will be overwritten if it is specified by the user.
> 
> To implement this, you most likely will need to create a `_helpers.tpl` file and make some helpers in it.
> 
> Have a look at what the helm chart that is generated when you run
> 
> ```
> helm create agent-stack-k8s
> ```
> 
> does.

How do you run 
```
helm create agent-stack-k8s
```
I made some changes and tried to run this for test, but got the error 
```
ERROR: The 'build' command is only supported from within a workspace (below a directory having a WORKSPACE file).
See documentation at https://bazel.build/concepts/build-ref#workspace
```
This repo do not have a workspace file...